plugins/out_stackdriver/stackdriver.c

```diff
@@ -761,7 +761,41 @@ static int get_stream(msgpack_object_map map)
     return STREAM_UNKNOWN;
 }
 
-static int pack_json_payload(int operation_extracted, int operation_extra_size, 
+static insert_id_status validate_insert_id(msgpack_object * insert_id_value, 
+                                           const msgpack_object * obj)                           
+{
+    int i = 0;
+    msgpack_object_kv * p = NULL;
+    insert_id_status ret = INSERTID_NOT_PRESENT;
+
+    if (obj == NULL) {
+        return ret;
+    }
+
+    for (i = 0; i < obj->via.map.size; i++) {
+        p = &obj->via.map.ptr[i];
+        if (p->key.type != MSGPACK_OBJECT_STR) {
+            continue;
+        }
+        if (p->key.via.str.size == INSERT_ID_SIZE 
```

Comment:
Side note: if https://github.com/fluent/fluent-bit/pull/2347 is merged please use the helper function implmeneted in https://github.com/fluent/fluent-bit/pull/2347 for key validation.

Reply:
Updated

Reply:
I don't think it is updated yet. I meant calling `validate_key` for this function.

Reply:
Updated!